airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/MetricQueries.java

```diff
@@ -153,43 +153,63 @@ public static List<Pair<JobStatus, Double>> overallJobRuntimeForTerminalJobsInLa
    * are not matching with the number of expected runs according to the schedule settings. Refer to
    * playbook for detailed discussion.
    */
-  public static Long numOfJobsNotRunningOnSchedule(final DSLContext ctx) {
+  public static Long percentOfJobsNotRunningOnSchedule(final DSLContext ctx) {
     final var countField = "cnt";
     final var query = """
-                      SELECT count(1) as cnt FROM ((
-                      	SELECT
+                      with abnormal_sync_jobs as
+                      (
+                      select count(1) as cnt
+                      from
+                      	(
+                      	select
                       		c.id,
                       		count(*) as cnt
-                      	FROM
+                      	from
                       		connection c
-                      	LEFT JOIN Jobs j ON j.scope::uuid = c.id
-                      	WHERE
-                      		c.schedule IS NOT null
-                      		AND c.schedule != 'null'
-                      		AND j.created_at > now() - interval '24 hours 1 minutes'
-                      		AND c.status = 'active'
-                      		AND j.config_type = 'sync'
-                      		AND c.updated_at < now() - interval '24 hours 1 minutes'
-                      		AND cast(c.schedule::jsonb->'timeUnit' as text) = '"hours"'
-                      	GROUP BY 1
-                      	HAVING count(*) < 24 / cast(c.schedule::jsonb->'units' as integer))
-                      UNION (
-                      SELECT
-                      	c.id,
-                      	count(*) as cnt
-                      FROM connection c
-                      LEFT JOIN Jobs j ON j.scope::uuid = c.id
-                      WHERE
-                      	c.schedule IS NOT null
-                      	AND c.schedule != 'null'
-                      	AND j.created_at > now() - interval '1 hours 1 minutes'
-                      	AND c.status = 'active'
-                      	AND j.config_type = 'sync'
-                      	AND c.updated_at < now() - interval '1 hours 1 minutes'
-                      	AND cast(c.schedule::jsonb->'timeUnit' as text) = '"minutes"'
-                      GROUP BY 1
-                      HAVING count(*) < 60 / cast(c.schedule::jsonb->'units' as integer))) as abnormal_sync_jobs
-                      	""";
+                      	left join Jobs j on
+                      		j.scope::uuid = c.id
+                      	where
+                      		c.schedule is not null
+                      		and c.schedule != 'null'
+                      		and j.created_at > now() - interval '24 hours 1 minutes'
```

Comment:
nit: the time fields can be pulled out to avoid repeats.

Reply:
how would you do that in SQL? Or you mean in Java style ( like ```"WHERE ... " + timeClauseConstant + " ... "```)

Reply:
I was referring to https://github.com/airbytehq/airbyte/blob/master/airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/MetricQueries.java#L85-L99